src/subscription_manager/repolib.py

```diff
@@ -781,6 +781,12 @@ def from_ent_cert_content(cls, content, baseurl, ca_cert, release_source):
         else:
             repo['gpgkey'] = utils.url_base_join(baseurl, gpg_url)
             # Leave gpgcheck as the default of 1
+        repomd_gpg_url = conf['rhsm']['repomd_gpg_url']
+        if repomd_gpg_url and repomd_gpg_url not in repo['gpgkey']:
+            if repo['gpgkey'] in ['', 'https://', 'http://']:
+                repo['gpgkey'] = repomd_gpg_url
+            else:
+                repo['gpgkey'] += ','+repomd_gpg_url
```

Comment:
Hmmm...  Moving this logic will fix the problem where I wasn't setting `gpgcheck` to `1` when the entitlement has no gpg key but repomd_gpg_url is set.

However, if we move this logic, and a full URL is provided for the repomd_gpg_url, but a relative path is listed for the entitlement's gpg key, then `url_base_join()` will not properly turn the entitlement gpgkey URL into an absolute URL ... Because repomd_gpg_url and gpgkey will be merged before the call to `url_base_join()`, so `url_base_join()` will detect the full repomd_gpg_url and skip prepending the baseurl:
https://github.com/candlepin/subscription-manager/blob/master/src/subscription_manager/utils.py#L117-L118

It looks like my change to the mutability logic was based on a misunderstanding of how server_value_repo was handled.  Now that I'm reading it again, I think that change was unnecessary regardless of the positioning of this logic.

I'll take another stab at this...